components/channel_layout/channel_identifier_router/actions.ts

```diff
@@ -38,7 +38,18 @@ export function onChannelByIdentifierEnter({match, history}: MatchAndHistory): A
             return {data: undefined};
         }
 
-        const channelPath = getPathFromIdentifier(state, path, identifier);
+        let channelPath = getPathFromIdentifier(state, path, identifier);
+
+        // A channel name that is exactly 26 characters long is recognized by
+        // getPathFromIdentifier as a 'channel_id'. So, if the channel_id does
+        // not exist, try it as a name. See
+        // https://mattermost.atlassian.net/browse/MM-32912.
+        if (channelPath === 'channel_id') {
```

Comment:
Could this be done in `getPathFromIdentifier` so that we could add a unit test for it?

Reply:
Actually, looking at `getPathFromIdentifier`, we already have a check there to see if a 26 character identifier is an ID or name, but it's looking for a channel with the given name and giving up if it can't find one. Perhaps it's not looking at the correct team since the repro steps for this bug involve multiple teams?